### PR TITLE
Windows Continuous Integration

### PR DESCRIPTION
**Pull request by chgibb**
Hello all,

I stumbled upon this project while trying to search for a way to compile Typescript to wasm. Turboscript seems close enough to Typescript to be of great use to me and therefore I would like to give back.

This PR will add CI support for Windows through appveyor. https://www.appveyor.com/
Please feel free to delete if Windows support is not a goal of this project.

I translated the current .travis.yml into the equivalent appveyor.yml. However all the tests are failing with the following.

```
npm run test
> turboscript@1.0.9-alpha test C:\projects\turboscript
> bash scripts/test
 FAIL  test\imports.spec.ts
  ● it should compile without error
    Compile Error! 
    TURBO_PATH:C:\projects\turboscript
      
      
      at Object.getWasmInstanceSync (test\utils\utils.ts:24:15)
      at Object.<anonymous>.test (test\imports.spec.ts:8:16)
  ● it should import sin function from javascript
    TypeError: Cannot read property 'exports' of undefined
      
      at Object.<anonymous>.test (test\imports.spec.ts:19:36)
  ● it should import cos function from javascript
    TypeError: Cannot read property 'exports' of undefined
      
      at Object.<anonymous>.test (test\imports.spec.ts:27:36)
  ● it should import tan function from javascript
    TypeError: Cannot read property 'exports' of undefined
      
      at Object.<anonymous>.test (test\imports.spec.ts:35:36)
  ● it should import abs function from javascript
    TypeError: Cannot read property 'exports' of undefined
      
      at Object.<anonymous>.test (test\imports.spec.ts:43:36)
 FAIL  test\array.spec.ts
  ● it should create an array
    Compile Error! 
    TURBO_PATH:C:\projects\turboscript
      
      
      at Object.getWasmInstanceSync (test\utils\utils.ts:24:15)
      at Object.<anonymous>.test (test\array.spec.ts:5:44)
 FAIL  test\linker.spec.ts
  ● it should compile without error
    Compile Error! 
    TURBO_PATH:C:\projects\turboscript
      
      
      at Object.getWasmInstanceSync (test\utils\utils.ts:24:15)
      at Object.<anonymous>.test (test\linker.spec.ts:8:19)
  ● it should call linked function externalFn
    TypeError: Cannot read property 'exports' of undefined
      
      at Object.<anonymous>.test (test\linker.spec.ts:20:32)
 FAIL  test\addTwo.spec.ts
  ● it should add two numbers
    Compile Error! 
    TURBO_PATH:C:\projects\turboscript
      
      
      at Object.<anonymous> (test\utils\utils.ts:42:15)
          at Generator.next (<anonymous>)
      at test\utils\utils.ts:7:71
          at Promise (<anonymous>)
      at Object.<anonymous>.__awaiter (test\utils\utils.ts:3:12)
      at Object.getWasmInstance (test\utils\utils.ts:46:12)
      at Object.<anonymous> (test\addTwo.spec.ts:5:49)
          at Generator.next (<anonymous>)
      at test\addTwo.spec.ts:7:71
          at Promise (<anonymous>)
      at Object.<anonymous>.__awaiter (test\addTwo.spec.ts:3:12)
      at Object.<anonymous>.test (test\addTwo.spec.ts:4:35)
Test Suites: 4 failed, 4 total
Tests:       9 failed, 9 total
Snapshots:   0 total
Time:        4.029s
Ran all test suites.
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "test"
npm ERR! node v4.8.1
npm ERR! npm  v2.15.11
npm ERR! code ELIFECYCLE
npm ERR! turboscript@1.0.9-alpha test: `bash scripts/test`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the turboscript@1.0.9-alpha test script 'bash scripts/test'.
npm ERR! This is most likely a problem with the turboscript package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     bash scripts/test
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs turboscript
npm ERR! Or if that isn't available, you can get their info via:
npm ERR! 
npm ERR!     npm owner ls turboscript
npm ERR! There is likely additional logging output above.
npm ERR! Please include the following file with any support request:
npm ERR!     C:\projects\turboscript\npm-debug.log
```

I'll have access to a Windows machine in a bit to try to debug further. In the meantime, can anyone shed some light as to why these tests might be failing?

### CODE REVIEW

**Comment by nidin** [comment]
Hi Chris,

I am glad that you find this project useful. Sure we are targeting all major platforms. Windows is one of it. I will check why it's failing.

PS: we have a slack team for casual discussions.
You may also open issues for public discussions.

Best,
Nidin

**Comment by chgibb** [comment]
Would I be able to join the Slack team?

**Comment by nidin** [comment]
Create an issue with your email id.

**Comment by nidin** [comment]
Can you try with yarn. 
yarn build
yarn test

**Comment by chgibb** [comment]
Still failing with the same errors.

**Comment by nidin** [comment]
Looks like appveyor's default node not ES6. default parameters is not supporting.

**Comment by chgibb** [comment]
Build is succeeding now https://ci.appveyor.com/project/chgibb/turboscript/history

File: package.json [hunk 1]
```diff
@@ -32,7 +33,7 @@
     "install:node-v8": "ts-node ./scripts/install-node-v8.ts",
     "build": "tsc -p ./src && tsc lib/tc.ts",
     "watch-build": "tsc -w -p ./src && tsc -w -p ./lib",
-    "test": "./node-v8/bin/node ./node_modules/jest/bin/jest.js ",
+    "test": "bash scripts/test",
```
Comment (nidin) [inline package.json]: Can you revert this line since bash is an additional dependency!

Comment (chgibb) [inline package.json]: Yarn test does not work on all platforms. Nor does npm test. It's fine for Linux/OSX but not Windows

File: test/utils/utils.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import * as path from "path";
 let spawnSync: any = child.spawnSync;
 if (process.platform === 'win32') {
     spawnSync = (cmd: string, args: string[]) => {
-        return child.spawnSync('node', [cmd].concat(args));
+        return child.spawnSync(path.join(__dirname, '../../node-v8/bin/node.exe'), [cmd].concat(args));
```
Comment (MaxGraey) [inline test/utils/utils.ts]: For construction like this path.join(__dirname, '../../<some path>') better using path.resolve

Comment (MaxGraey) [inline test/utils/utils.ts]: For paths like this better use path.resolve instead path.join

File: package.json (hunk 1 above)
Comment (nidin) [inline package.json]: That's not true `yarn test` should work in all platforms. It's working on my windows machine and appveyor https://ci.appveyor.com/project/nidin/turboscript/build/1.0.8

Comment (nidin) [inline package.json]: Can you try it on your fork if it is working on appveyor.

Comment (chgibb) [inline package.json]: My bad then. I'll be able to make and test the change tomorrow.

**Comment by chgibb** [comment]
I made the requested changes. I did not know that would work with Yarn. That is fantastic! I've been burned way too many times by that syntax with npm scripts. I might just move my projects to Yarn...

**Pull request merged**